### PR TITLE
Add Workspace Commands and Tests

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

This PR adds the commands astro workspace create, update, and delete

## 🎟 Issue(s)

- https://github.com/astronomer/astro-cli/issues/1131

## 🧪 Functional Testing

manual testing

## 📸 Screenshots

Screenshot 2023-03-29 at 3 57 06 PM

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cloud/workspace/workspace.go [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+	"github.com/astronomer/astro-cli/cloud/user"
 	"github.com/astronomer/astro-cli/config"
+	"github.com/astronomer/astro-cli/context"
 	"github.com/astronomer/astro-cli/pkg/input"
 	"github.com/astronomer/astro-cli/pkg/printutil"
 )
 
-var errInvalidWorkspaceKey = errors.New("invalid workspace selection")
+var (
+	errInvalidWorkspaceKey = errors.New("invalid workspace selection")
```
Comment (kushalmalani) [inline cloud/workspace/workspace.go:22]: is this used? looks to be same as `ErrInvalidWorkspaceKey`

File: cmd/cloud/workspace.go [hunk 2]
```diff
@@ -60,6 +66,53 @@ func newWorkspaceSwitchCmd(out io.Writer) *cobra.Command {
 	return cmd
 }
 
+func newWorkspaceCreateCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{"cr"},
+		Short:   "create an Astro Workspace",
```
Comment (kushalmalani) [inline cmd/cloud/workspace.go:73]: Can we capitalize the first letter like we do for List and Switch

Comment (kushalmalani) [reply cmd/cloud/workspace.go:73]: Same for Update and Delete

File: cmd/cloud/workspace.go [hunk 3]
```diff
@@ -60,6 +66,53 @@ func newWorkspaceSwitchCmd(out io.Writer) *cobra.Command {
...
+	cmd.Flags().StringVarP(&workspaceName, "name", "n", "", "The Workspace's name. If the name contains a space, specify the entire name within quotes \"\" ")
+	cmd.Flags().StringVarP(&workspaceDescription, "description", "d", "", "Description of the Workspace. If the description contains a space, specify the entire workspace in quotes \"\"")
+	cmd.Flags().StringVarP(&enforceCD, "enforce-cicd", "e", "OFF", "Provide this flag either ON/OFF. ON means deploys to deployments must use an API Key or Token. This essentially forces Deploys to happen through CI/CD")
+	return cmd
+}
+
+func newWorkspaceDeleteCmd(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "delete [workspace_id]",
+		Aliases: []string{"up"},
```
Comment (kushalmalani) [inline cmd/cloud/workspace.go:105]:
```suggestion
		Aliases: []string{"de"},
```

**Review by kushalmalani — APPROVED** [review]
Changes look good. Left a couple of nit comments. Also, can your rebase and bump the code coverage

File: cloud/workspace/workspace.go (hunk 1 above)
Comment (sunkickr) [reply cloud/workspace/workspace.go:22]: yeah ErrInvalidWorkspaceKey is not needed

**Pull request merged**